src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,85 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
+    private static final Map<Class<?>, Class<?>> interfaceImplementations = Map.of(
+            ProductService.class, ProductServiceImpl.class,
+            ProductParser.class, ProductParserImpl.class,
+            FileReaderService.class, FileReaderServiceImpl.class
+    );
+
     private static final Injector injector = new Injector();
 
+    private final Map<Class<?>, Object> instances = new HashMap<>();
+
+    private Injector() {
+    }
+
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Object clazzImplementation = null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        if (!clazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException(
+                    "Injection failed, missing @Component annotation on the class "
+                            + clazz.getName());
+        }
+        Field[] declaredFields = clazz.getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = getInstance(field.getType());
+                clazzImplementation = createNewInstance(clazz);
+                setFieldValue(clazzImplementation, field, fieldInstance);
+            }
+        }
+        if (clazzImplementation == null) {
+            clazzImplementation = createNewInstance(clazz);
+        }
+        return clazzImplementation;
+    }
+
+    private Class<?> findImplementation(Class<?> interfaceClazz) {
+        if (interfaceClazz.isInterface()) {
+            return interfaceImplementations.get(interfaceClazz);
```

Comment:
let's also throw an runtime exception if the implementation for the interface was not found, e.g.:
```
 if (interfaceImplementations.get(interfaceClazz) == null) {
               throw new RuntimeException("The implementation for the interface"
                    + interfaceClazz.getName() + " was not found.");
           }
```